src/page/AiPage.tsx

```diff
@@ -0,0 +1,199 @@
+  const openai = new OpenAI({
+    apiKey: process.env.REACT_APP_APIKEY_OPENAI,
+    dangerouslyAllowBrowser: true
+  });
+
+  const onChangeMessage = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setQuestionChatGPT(event.target.value);
+  };
+
+  async function chatGPTHandler() {
+    const newController = new AbortController();
+    setAbortController(newController);
+
+    if (isWaitResponse) {
+      newController.abort();
+      setIsWaitResponse(false);
+      return;
+    }
+
+    setIsWaitResponse(true);
+    setChatHistory((prevState) => [...prevState, { role: 'user', content: questionChatGPT }]);
+
+    const myMessages = chatHistory.map((message) => ({
+      role: message.role,
+      content: message.content
+    })) as ChatCompletionMessageParam[];
+
+    myMessages.push({
+      role: 'user',
+      content: questionChatGPT
+    });
+    setQuestionChatGPT('');
+
+    try {
+      const completion = await openai.chat.completions.create(
+        {
+          messages: [...myMessages],
+          model: 'gpt-3.5-turbo'
+        },
+        { signal: newController.signal }
+      );
+
+      if (completion.choices[0].message.content !== null) {
+        setChatHistory((prevState) => [
+          ...prevState,
+          { role: 'assistant', content: completion.choices[0].message.content! }
+        ]);
+      }
+    } catch (error) {
+      console.log(error);
+    } finally {
+      setIsWaitResponse(false);
+    }
+  }
+
+  const handleKeyPress = (event: React.KeyboardEvent) => {
+    if (event.key === 'Enter') {
+      event.preventDefault();
+      chatGPTHandler();
+    }
+  };
+
+  const clickHandler = () => {
+    if (isWaitResponse) {
+      abortController!.abort();
+      setIsWaitResponse(false);
+      return;
+    }
+
+    chatGPTHandler();
+  };
+
+  const backProjectHandler = () => {
+    history(`/project/${idActiveProject}`);
+  };
+
+  const todoClickHandler = (todo: {
+    status: string;
+    title: string;
+    description: string;
+    importance: string;
+    id: string;
+    idProject: string;
+    Labels: string[];
+  }) => {
+    setQuestionChatGPT(todo.description);
+  };
+
+  useEffect(() => {
+    const projectActiveId = localStorage.getItem('projectActiveId')!;
+    console.log(projectActiveId, '111');
+    console.log(todosProject);
+    if (projectActiveId) {
+      setIdActiveProject(projectActiveId);
+    }
+  }, []);
+  useEffect(scrollToBottom, [chatHistory]);
+
+  return (
+    <AiPageWrapper>
+      <AiSidebarWrapper>
+        <div
+          onClick={backProjectHandler}
+          className="px-2 py-2 hover:bg-blue-700 rounded mb-4 cursor-pointer "
+        >
+          <BackArrow icon={faArrowLeft} />
+          Back to project
+        </div>
+        {todosProject.map((todo, index) => {
+          return (
+            <div
+              key={index}
+              className="px-2 py-2 hover:bg-blue-700 rounded cursor-pointer"
+              onClick={() => todoClickHandler(todo)}
```

Comment:
let's create seperate component for this mapping and create new handler for onClick